tutor/src/screens/course-settings/course-details.js

```diff
@@ -145,9 +145,12 @@ const CourseDetails = observer(({ course, history }) => {
 
     const onSubmit = ({ courseName, courseCode, timezone }) => {
         setIsSaving(true);
-        course.name = courseName;
-        course.code = courseCode;
-        course.timezone = timezone;
+        runInAction(() => {
+            course.name = courseName;
+            course.code = courseCode;
+            course.timezone = timezone;
+        })
```

Comment:
I think you can make `onSubmit` be an action and then you wouldn't need the runInAction.  The only time you can't do that is if the method is async and the changes occur after the 1st await statement.  In that case babel compiles the remainder of the method to be called in a separate function which isn't decorated as an action so you have to do it yourself